### PR TITLE
[Merged by Bors] - feat(Analysis/SpecialFunctions/Exp): cexp is uniformlyContinuousOn

### PR DESCRIPTION
**Pull request by CBirkbeck**
We show that the complex exponential is uniformly continuous on left half planes.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/SpecialFunctions/Exp.lean [hunk 1]
```diff
@@ -117,6 +117,37 @@ theorem ContinuousOn.cexp (h : ContinuousOn f s) : ContinuousOn (fun y => exp (f
...
+  have ha : 0 < ε / (2 * Real.exp a) := by positivity
+  have H := this 0 (ε / (2 * Real.exp a)) ha
+  rw [Metric.eventually_nhds_iff] at H
+  obtain ⟨δ, hδ⟩ := H
+  refine ⟨δ, hδ.1, ?_⟩
+  intros x _ y hy hxy
+  have h3 := hδ.2 (y := x - y) (by simpa only [dist_zero_right, norm_eq_abs] using hxy)
+  rw [dist_eq_norm, exp_zero] at *
+  have : cexp x - cexp y = cexp y * (cexp (x - y) - 1) := by
+      rw [@mul_sub_one, ← exp_add]
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/SpecialFunctions/Exp.lean:136]: Is the @ needed?

File: Mathlib/Analysis/SpecialFunctions/Exp.lean [hunk 2]
```diff
@@ -117,6 +117,37 @@ theorem ContinuousOn.cexp (h : ContinuousOn f s) : ContinuousOn (fun y => exp (f
 theorem Continuous.cexp (h : Continuous f) : Continuous fun y => exp (f y) :=
   continuous_iff_continuousAt.2 fun _ => h.continuousAt.cexp
 
+/--The complex exponential function is uniformly continuous on left half planes. -/
+lemma UniformlyContinuosOn_cexp (a : ℝ) : UniformContinuousOn exp {x : ℂ | x.re ≤ a} := by
+  have : Continuous (cexp - 1) := Continuous.sub (Continuous.cexp continuous_id') continuous_one
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/SpecialFunctions/Exp.lean:122]: Might be more readable with dot notation

**Review by Ruben-VandeVelde — APPROVED** [review]

File: Mathlib/Analysis/SpecialFunctions/Exp.lean (hunk 1 above)
Comment (CBirkbeck) [reply Mathlib/Analysis/SpecialFunctions/Exp.lean:136]: no that was just copilot! I didnt notice it :)

File: Mathlib/Analysis/SpecialFunctions/Exp.lean [hunk 3]
```diff
@@ -117,6 +117,37 @@ theorem ContinuousOn.cexp (h : ContinuousOn f s) : ContinuousOn (fun y => exp (f
 theorem Continuous.cexp (h : Continuous f) : Continuous fun y => exp (f y) :=
   continuous_iff_continuousAt.2 fun _ => h.continuousAt.cexp
 
+/--The complex exponential function is uniformly continuous on left half planes. -/
```
Comment (dupuisf) [inline Mathlib/Analysis/SpecialFunctions/Exp.lean:120]:
```suggestion
/-- The complex exponential function is uniformly continuous on left half planes. -/
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Pull request closed without merging**